soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -2697,46 +2697,105 @@ CustomMessage Randomizer::GetMerchantMessage(RandomizerCheck rc, TextIDs textId,
     return messageEntry;
 }
 
+bool Randomizer::IsQuestKnown(RandomizerCheckArea area) {
+    switch (area) {
+        case RCAREA_DEKU_TREE:
+            return GetRandoSettingValue(RSK_MQ_DEKU_TREE) != RO_MQ_SET_RANDOM;
+        case RCAREA_DODONGOS_CAVERN:
+            return GetRandoSettingValue(RSK_MQ_DODONGOS_CAVERN) != RO_MQ_SET_RANDOM;
+        case RCAREA_JABU_JABUS_BELLY:
+            return GetRandoSettingValue(RSK_MQ_JABU_JABU) != RO_MQ_SET_RANDOM;
+        case RCAREA_FOREST_TEMPLE:
+            return GetRandoSettingValue(RSK_MQ_FOREST_TEMPLE) != RO_MQ_SET_RANDOM;
+        case RCAREA_FIRE_TEMPLE:
+            return GetRandoSettingValue(RSK_MQ_FIRE_TEMPLE) != RO_MQ_SET_RANDOM;
+        case RCAREA_WATER_TEMPLE:
+            return GetRandoSettingValue(RSK_MQ_WATER_TEMPLE) != RO_MQ_SET_RANDOM;
+        case RCAREA_SPIRIT_TEMPLE:
+            return GetRandoSettingValue(RSK_MQ_SPIRIT_TEMPLE) != RO_MQ_SET_RANDOM;
+        case RCAREA_SHADOW_TEMPLE:
+            return GetRandoSettingValue(RSK_MQ_SHADOW_TEMPLE) != RO_MQ_SET_RANDOM;
+        case RCAREA_BOTTOM_OF_THE_WELL:
+            return GetRandoSettingValue(RSK_MQ_BOTTOM_OF_THE_WELL) != RO_MQ_SET_RANDOM;
+        case RCAREA_ICE_CAVERN:
+            return GetRandoSettingValue(RSK_MQ_ICE_CAVERN) != RO_MQ_SET_RANDOM;
+        case RCAREA_GERUDO_TRAINING_GROUND:
+            return GetRandoSettingValue(RSK_MQ_GTG) != RO_MQ_SET_RANDOM;
+        case RCAREA_GANONS_CASTLE:
+            return GetRandoSettingValue(RSK_MQ_GANONS_CASTLE) != RO_MQ_SET_RANDOM;
+        default:
+            return true;
+    }
+}
+
+bool Randomizer::AreQuestsVaried() {
+    u8 dekuValue = GetRandoSettingValue(RSK_MQ_DEKU_TREE);
+    for (int i = (int)RSK_MQ_DEKU_TREE; i <= (int)RSK_MQ_GANONS_CASTLE; i++) {
+        u8 value = GetRandoSettingValue((RandomizerSettingKey)i);
+        if (value == (u8)RO_MQ_SET_RANDOM || dekuValue != value) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool Randomizer::QuestCompatible(RandomizerCheckQuest quest, RandomizerCheckArea area) {
```

Comment:
```suggestion
bool Randomizer::QuestActiveForArea(RandomizerCheckQuest quest, RandomizerCheckArea area) {
```
not sure what to name this